lib/node.js

```diff
@@ -192,18 +191,33 @@ Node.prototype._deleteRecursive = function (key, distinct, index) {
     return this;
   }
 
+  if (index === key.length) {
+    if (this.children[undefined]) {
+      this.children[undefined]._deleteRecursive(key, distinct, index + 1);
+    }
```

Comment:
This is to resolve the initial edge case you mentioned with `undefined` keys, right?

Reply:
Yes exactly. 
the `undefined` key is used for non-leaf nodes to identify that [this item is exactly this key]


Visually: 
```
// skipping "children" and "values" for brevity
{
   a: {
       undefined: {}  we have an item with key *exactly* "a" and nothing longer
       a: {...} // we have an item with prefix "aa", but don't yet know if its shorter or not (need to look deeper)
   }
   b: {
      b: {
         undefined: {}  we have an item with key *exactly* "bb" and not longer
         a: {...} // we have an item with prefix "bba", but don't yet know if its shorter or not (need to look deeper)
      }
   }
}
```

Reply:
Thanks for the example, that makes perfect sense!